Client/src/main/java/org/gluu/oxauth/client/TokenClient.java

```diff
@@ -252,7 +276,8 @@ public TokenResponse exec() {
         if (StringUtils.isNotBlank(getRequest().getRefreshToken())) {
             clientRequest.formParameter("refresh_token", getRequest().getRefreshToken());
         }
-        if (getRequest().getAuthenticationMethod() == AuthenticationMethod.CLIENT_SECRET_POST) {
+        if (getRequest().getAuthenticationMethod() == AuthenticationMethod.CLIENT_SECRET_POST ||
+                getRequest().getAuthenticationMethod() == AuthenticationMethod.TLS_CLIENT_AUTH) {
```

Comment:
Good but I guess we also need `SELF_SIGNED_TLS_CLIENT_AUTH` here.

Reply:
You're right, I haven't had the change to use this way to do this, but I'll test it using oxTrust also. Thanks.